doc/methods/reprocessing.rst

```diff
@@ -3,4 +3,162 @@
+    
+:class:`saltproc.Materialflow`. objects are used to store information about
+materials in reprocessing as well as materials being fed into the system.
+
+Processes
+.........
+A process is an abstraction of chemical separation, extraction, or some
+other removal. Two processes commonly used in \Gls{msr}s include
+extracting metals using gas sparging and filtering. In SaltProc,
+:class:`saltproc.Process` objects contain data and functions to perform their
+associated processing task. At a minimum, a :class:`saltproc.Process` object includes
+the following
+
+1. A mass flow rate, :math:`\dot{m}`, which specifies the mass of fuel salt a process can operate on per unit time
+2. An extraction efficiency, :math:`\epsilon`, for target element(s).
+3. A method to apply the process on a :class:`saltproc.Materialflow` object. 
+
+Graphs
+......
+A graph is a mathematical object that connects *nodes*
+with *edges*. More formally,
+a graph is a pair of sets, :math:`(V, E)`, where
+
+- :math:`V` is a set whose elements are called nodes
+- :math:`E` is a set whose elements are pairs of elements in :math:`V`
+
+A directed graph is a graph in which the elements are ordered pairs of elements
+in :math:`V`. This gives the edges a direction. SaltProc uses directed graphs to
+model the order and path in which processes operate on materials.
+        
+Processes and feeds are defined in one JSON input file, and the graph linking
+processes is defined in a DOT file. The process graph must be directed and
+acyclic in order to work with SaltProc. At runtime,
+SaltProc reads this input file to create :class:`saltproc.Process` objects for each item
+in the file. Every process file must have at least a :class:`core_outlet` and
+:class:`core_inlet` Process.
+
+Material reprocessing
+---------------------
+
+Recall that SaltProc uses a *batchwise* reprocessing scheme.
+Let :math:`\mathbf{n}(t)^{j}` denote the nuclide mass vector for depletable material
+:math:`j` as a function of time. For each depletable material :math:`j`, the depletion
+solver numerically integrates the equation
+
+.. math::
+
+    \frac{d\mathbf{n}^{j}(t)}{dt} = \mathbf{A}(\mathbf{n}^{j}(t), t)
+
+from time :math:`i` to time :math:`i+1` to get :math:`\mathbf{n}^{j}(i+1)`, where :math:`A` is the
+depletion matrix. This sequence is called a **depletion step**.
+
+Now, let the mass and volume of depletable material :math:`j` be
+:math:`m^{j}` and :math:`V^{j}` respectively. Let the mass flow rate of process :math:`p` be
+:math:`\dot{m}_{p}`. At the end of each depletion step, SaltProc constructs process
+paths from the process graph defined in the DOT file, and sequentially applies
+each process :math:`p` in each path :math:`r` to the relevant materials to obtain through
+and waste streams for each material. SaltProc tracks the mass and nuclide vector for both through and waste streams. For through
+streams, SaltProc also tracks the volume and mass flow rate. For every node
```

Comment:
"For" can be removed here I think, unless you mean `through streams`. Consider changing to `through-streams` for readability